zebra-test/src/command.rs

```diff
@@ -3,33 +3,40 @@ use color_eyre::{
     Help, SectionExt,
 };
 use std::process::{Child, Command, ExitStatus, Output};
-use std::{fs, io::Write};
+use std::{fs, io::Write, path::PathBuf};
 use tempdir::TempDir;
 
 #[cfg(unix)]
 use std::os::unix::process::ExitStatusExt;
 
-/// Runs a command in a TempDir
-pub fn test_cmd(path: &str) -> Result<(Command, impl Drop)> {
-    let dir = TempDir::new(path)?;
-    let mut cmd = Command::new(path);
-    cmd.current_dir(dir.path());
-
-    let cache_dir = dir.path().join("state");
-    fs::create_dir(&cache_dir)?;
-
-    fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
-        format!(
-            "[state]\ncache_dir = '{}'",
-            cache_dir
-                .into_os_string()
-                .into_string()
-                .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
-        )
-        .as_bytes(),
-    )?;
-
-    Ok((cmd, dir))
+/// Runs a command
+pub fn test_cmd(command_path: &str) -> Result<Command> {
+    let mut cmd = Command::new(command_path);
+    cmd.current_dir(PathBuf::from(command_path).parent().unwrap().to_path_buf());
+
+    Ok(cmd)
+}
+
+/// Create a temp directory and optional config file
+pub fn tempdir(create_config: bool) -> Result<(PathBuf, impl Drop)> {
```

Comment:
Given that this pathbuf is usually ignored we should probably just return a `TempDir` instead of the `impl Drop` and let the tests get the path when they need it from the `TempDir` itself.

Reply:
By the changes done at https://github.com/ZcashFoundation/zebra/pull/859/commits/a27e5a243fb98084cf6ee8ce7852113661e5a9c5 the pathbuf is now actually used. Not sure if we still want to make changes here.